auth/maintenance_socket_role_manager.cc

```diff
@@ -109,4 +110,8 @@ future<> maintenance_socket_role_manager::remove_attribute(std::string_view role
     return operation_not_supported_exception("REMOVE ATTRIBUTE");
 }
 
+future<std::vector<cql3::description>> maintenance_socket_role_manager::describe_role_grants() {
+    return operation_not_supported_exception<std::vector<cql3::description>>("DESCRIBE SCHEMA WITH INTERNALS");
```

Comment:
How does a an operation_not_supported_exception\<whatever> convert to future\<std::vector>?

Reply:
It's a function, not an object:

```cpp
template<typename T = void>
future<T> operation_not_supported_exception(std::string_view operation) {
    return make_exception_future<T>(
        std::runtime_error(fmt::format("role manager: {} operation not supported through maintenance socket", operation)));
}
```

It's a bit weird, but I used it to stay consistent with the rest of the file. If it should be ditched/changed to something else, we can do it in a separate PR.

Reply:
It's badly named (functions should be verbs, not object). Please follow up with a cleanup.